packages/components/ovh-shell/__tests__/plugin/auth/auth.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { loadFeature, defineFeature } from 'jest-cucumber';
+import auth from '../../../src/plugin/auth';
+
+const feature = loadFeature('../../../features/plugin/auth/auth.feature', {
+  loadRelativePath: true,
+});
+
+defineFeature(feature, (test) => {
+  let authPlugin;
+  delete window.location;
+  const origin = 'https://www.ovh.com';
+  const href = `${origin}/manager/#/`;
+  window.location = {
+    assign: jest.fn(),
+    href,
+    origin,
+  };
+
+  test('Log in', ({ given, when, then }) => {
+    given('An authentication plugin', () => {
+      authPlugin = auth();
+    });
+
+    when('I try to log in', () => {
+      authPlugin.login();
+    });
+
+    then(
+      'I should get redirected to the authentication page with login parameters',
+      () => {
+        expect(window.location.assign).toHaveBeenCalledWith(
+          `${origin}/auth/?onsuccess=${encodeURIComponent(href)}`,
+        );
+      },
+    );
+  });
+
+  test('Log in with a given subsidiary', ({ given, when, then }) => {
+    given('An authentication plugin', () => {
+      authPlugin = auth();
+    });
+
+    when('I try to log in with an ovh subsidiary', () => {
+      authPlugin.login({
+        ovhSubsidiary: 'foo',
+      });
+    });
+
+    then(
+      'I should get redirected to the authentication page with login prefilled',
+      () => {
+        expect(window.location.assign).toHaveBeenCalledWith(
+          `${origin}/auth/?ovhSubsidiary=foo&onsuccess=${encodeURIComponent(href)}`,
+        );
+      },
+    );
+  });
+
+  test('Log in with a login redirection URL', ({ given, when, then }) => {
+    const onLoginURL = 'http://www.ovh.com/manager/dedicated';
+
+    given('An authentication plugin', () => {
+      authPlugin = auth();
+    });
+
+    when('I try to log in with a redirection URL parameter', () => {
+      authPlugin.login({
+        onLoginURL,
+      });
+    });
+
+    then(
+      'I should get redirected to the authentication page with onsuccess parameter setted to the URL',
+      () => {
+        expect(window.location.assign).toHaveBeenCalledWith(
+          `${origin}/auth/?onsuccess=${encodeURIComponent(onLoginURL)}`,
+        );
+      },
+    );
+  });
+
+  test('Log out', ({ given, when, then }) => {
+    given('An authentication plugin', () => {
+      authPlugin = auth();
+    });
+
+    when('I try to log out', () => {
+      authPlugin.logout();
+    });
+
+    then(
+      'I should get redirected to the authentication page with logout parameters',
+      () => {
+        expect(window.location.assign).toHaveBeenCalledWith(
```

Comment:
This is not a reliable test. It tests too much the implementation details. We shouldn't test if `window.location.assign`is called with certain parameters, and we don't care that if it's called or not. 

Because it can change. 

But i know that DOM methods can be painful to test, but to get around that, we should split the code. 
We know that `window.location.assign`, will always call `addParameters`, so we should test the output of that method instead. So that we know that that method, is tested in a **unitary** way. 
And since you have logic in `login`/`logout`  that construct `params` based on `authURL` and `loginParameters`, you should extract that logic to test it separately, for example in a `constructAuthParameters(login/logout, LoginParams)` method.
And since these methods will be tested unitarily, you won't have to test the whole `login`/`logout` method. Which can actually be better tested with `e2e`tests. 

Or another solution, is to wrap your `window.location.assign` in another method, for example `redirect` and test that `redirect` is called with the parameters you want. But this solution should be your last resort if there is no another solution. So the above solution is better to have more reliable unit tests.